janitor/pkg/csp/nebius/nebius.go

```diff
@@ -0,0 +1,279 @@
+// Supports two authentication methods (similar to OCI):
+// 1. NEBIUS_IAM_TOKEN environment variable (direct token)
+// 2. NEBIUS_SA_KEY_FILE service account credentials file (recommended for production)
+func getSDK(ctx context.Context) (*gosdk.SDK, error) {
+	var creds gosdk.Credentials
+
+	// Check for direct IAM token first
+	if token := os.Getenv("NEBIUS_IAM_TOKEN"); token != "" {
+		creds = gosdk.IAMToken(token)
+	} else if keyFile := os.Getenv("NEBIUS_SA_KEY_FILE"); keyFile != "" {
+		// Use service account credentials file (recommended for production)
+		creds = gosdk.ServiceAccountReader(
+			auth.NewServiceAccountCredentialsFileParser(nil, keyFile),
+		)
+	} else {
+		return nil, fmt.Errorf("no Nebius credentials found: set NEBIUS_IAM_TOKEN or NEBIUS_SA_KEY_FILE")
+	}
+
+	sdk, err := gosdk.New(ctx, gosdk.WithCredentials(creds))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Nebius SDK: %w", err)
+	}
+
+	return sdk, nil
+}
+
+// getInstanceService returns the instance service, either from mock (for testing) or real SDK.
+func (c *Client) getInstanceService(ctx context.Context) (InstanceService, func(), error) {
+	// Use mock client if available (for testing)
+	if c.instanceService != nil {
+		return c.instanceService, func() {}, nil
+	}
+
+	// Production: create SDK and return instance service
+	sdk, err := getSDK(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Return cleanup function to close SDK
+	cleanup := func() { sdk.Close() }
+
+	return sdk.Services().Compute().V1().Instance(), cleanup, nil
+}
+
+// SendRebootSignal sends a reboot signal to Nebius for the given node.
+// Nebius doesn't have a direct reboot API, so we stop the instance first.
+// The instance will be started in IsNodeReady after a cooldown period.
+func (c *Client) SendRebootSignal(ctx context.Context, node corev1.Node) (model.ResetSignalRequestRef, error) {
+	logger := log.FromContext(ctx)
+
+	// Fetch the node's provider ID
+	providerID := node.Spec.ProviderID
+	if providerID == "" {
+		err := fmt.Errorf("no provider ID found for node %s", node.Name)
+		logger.Error(err, "Failed to reboot node")
+		return "", err
+	}
+
+	// Extract the instance ID from the provider ID
+	nodeFields, err := getNodeFields(node)
+	if err != nil {
+		logger.Error(err, "Failed to parse provider ID")
+		return "", err
+	}
+
+	logger.Info("Stopping Nebius instance for reboot", "instanceID", nodeFields.instanceID)
+
+	instanceService, cleanup, err := c.getInstanceService(ctx)
+	if err != nil {
+		logger.Error(err, "Failed to get Nebius instance service")
+		return "", err
+	}
+	defer cleanup()
+
+	op, err := instanceService.Stop(ctx, &compute.StopInstanceRequest{
+		Id: nodeFields.instanceID,
+	})
+	if err != nil {
+		logger.Error(err, "Failed to stop instance", "instanceID", nodeFields.instanceID)
+		return "", err
+	}
+
+	// Wait for the stop operation to complete
+	if _, err := op.Wait(ctx); err != nil {
```

Comment:
Can we make this aync? I.e., can we store the op in `model.ResetSignalRequestRef` and then use `IsNodeReady` for polling? Please see `janitor/pkg/csp/gcp/gcp.go`, we have done something similar there